lib/item_repo.rb

```diff
@@ -32,11 +32,60 @@ def all
   end
 
   def find_item_by_id(id)
-    name = []
+    id_number = []
     item_objects(@items_csv).each do |item|
-      name << item if item.id == id
+      id_number << item if item.id == id
     end
-    name[0]
+    id_number[0]
   end
 
+  def find_by_name(name)
+    item_name = []
+    item_objects(@items_csv).each do |row|
+      item_name << row if row.name == name
+    end
```

Comment:
i like the way you use the if statement in this syntax. it is much easier for a human to understand right away